### PR TITLE
Do not ignore revertOnFailure flag when validating Address/Contract

### PR DESCRIPTION
**Pull request by axic**
From #5815.

### CODE REVIEW

**Comment by axic** [comment]
Effectively this is a no-op because the flag is never set, yet.

File: libsolidity/codegen/ABIFunctions.cpp [hunk 1]
```diff
@@ -341,7 +341,7 @@ string ABIFunctions::conversionFunction(Type const& _from, Type const& _to)
 				solAssert(_from.mobileType(), "");
 				body =
 					Whiskers("converted := <cleanEnum>(<cleanInt>(value))")
-					("cleanEnum", cleanupFunction(_to, false))
```
Comment (chriseth) [inline libsolidity/codegen/ABIFunctions.cpp:344]: Why not pass along the flag?

Comment (axic) [reply libsolidity/codegen/ABIFunctions.cpp:344]: There's no flag in the "conversionFunction" to be passed around. It defaults to `false` which stands for "assert". I think that is what we want at encoding time.

Comment (chriseth) [reply libsolidity/codegen/ABIFunctions.cpp:344]: I hope we can use that function for general conversions. The decoder always verifies prior to conversion, right?

Comment (axic) [reply libsolidity/codegen/ABIFunctions.cpp:344]: What do you mean?

Comment (chriseth) [reply libsolidity/codegen/ABIFunctions.cpp:344]: I hope this conversion function can later be used for the solidity to yul code generator.

My question would be: If we assert in a failed conversion here. Might it be that the ABI decoder calls convert to enum (from whatever) without validating the range prior to calling convert? If not, we should revert instead.

Comment (axic) [reply libsolidity/codegen/ABIFunctions.cpp:344]: To be clear, this change here doesn't change the current semantics.

I'm not sure about all the possible code paths leading to convert.

**Review by chriseth — APPROVED** [review]

**Pull request merged**